### PR TITLE
Add GetMarker-Function and Display Marker (Backend + Frontend)

### PR DESCRIPTION
**Pull request by Cathesso**
When the user position is found, an area around the user is created.
That area is sent to the backend. The backend asks Overpass for certain Nodes in that Area.
Those Nodes are sent back from the backend to the frontend and then displayed.

### CODE REVIEW

File: backend/src/main/java/de/cathesso/scratchamap/controller/MapDataController.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+public class MapDataController {
+
+    private final MapDataService mapDataService;
+
+    @Autowired
+    public MapDataController(MapDataService mapDataService){this.mapDataService = mapDataService;}
+
+    @GetMapping("getmarkers")
+    public List<Marker> getMarkers(@RequestParam String sWLat, String sWLon, String nELat, String nELon){
+        System.out.println("DINGDINGDING");
```
Comment (christophersiem) [inline backend/src/main/java/de/cathesso/scratchamap/controller/MapDataController.java]:
```suggestion
```

File: backend/src/main/java/de/cathesso/scratchamap/model/Marker.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+@Document(collection="Markers")
+public class Marker {
+    @Id
+    private String id;
+    private String latitude;
+    private String longitude;
+    //private String type; <- To be added later. Information to be extracted from Tags (Streetlight etc.)
```
Comment (christophersiem) [inline backend/src/main/java/de/cathesso/scratchamap/model/Marker.java]:
```suggestion
```

File: backend/src/main/java/de/cathesso/scratchamap/service/OverpassApiService.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    private final String overpassApi = "https://overpass-api.de/api/interpreter?data=";
+    private final RestTemplate restTemplate;
+
+    @Autowired
+    public OverpassApiService(RestTemplate restTemplate) {this.restTemplate = restTemplate;}
+
+    public List<OverpassApiElement> getMapElements(String sWLat, String sWLon, String nELat, String nELon) {
+
+        String requestString = overpassApi + "[out:json][timeout:25];(way[\"highway\"][\"highway\"!=\"motorway\"][\"highway\"!=\"motorway_link\"][\"highway\"!=\"primary\"][\"highway\"!=\"service\"][\"foot\"!=\"no\"][\"access\"!=\"private\"][\"access\"!=\"no\"](" + sWLat + "," + sWLon + "," + nELat + "," + nELon + ");>;);out qt;";
+        System.out.println(requestString);
```
Comment (christophersiem) [inline backend/src/main/java/de/cathesso/scratchamap/service/OverpassApiService.java]:
```suggestion
```

File: backend/src/main/java/de/cathesso/scratchamap/service/OverpassApiService.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        return Marker.builder()
+                .id(overpassApiElement.getId())
+                .latitude(overpassApiElement.getLat())
+                .longitude(overpassApiElement.getLon())
+                .build();
+    }
+
+    public List<Marker> getMapNodes(String sWLat, String sWLon, String nELat, String nELon){
+        List<Marker> Nodes = getMapElements(sWLat, sWLon, nELat, nELon).stream().filter(element -> element.getType().equals("node")).map(this::mapOverpassNodesToMarkerModel).collect(Collectors.toList());
+        return Nodes;
```
Comment (christophersiem) [inline backend/src/main/java/de/cathesso/scratchamap/service/OverpassApiService.java]:
```suggestion
        return getMapElements(sWLat, sWLon, nELat, nELon).stream().filter(element -> element.getType().equals("node")).map(this::mapOverpassNodesToMarkerModel).collect(Collectors.toList());
```

File: frontend/src/components/StreetMap.js [hunk 5]
```diff
@@ -1,29 +1,84 @@
...
+let nodeIcon = L.icon({
+  iconUrl: "https://image.flaticon.com/icons/png/128/2/2013.png",
+  iconSize: [10, 10],
+});
+
 export default function StreetMap() {
   let userMarker = new L.marker([27.380583, 33.631839], { icon: userIcon });
+  const [markers, setMarkers] = useState([]);
+  const [userBounds, setUserBounds] = useState("");
+  //const [userIsWithinBounds, setUserIsWithinBounds] = useState(false);
```
Comment (christophersiem) [inline frontend/src/components/StreetMap.js]:
```suggestion
```

File: frontend/src/components/StreetMap.js [hunk 6]
```diff
@@ -1,29 +1,84 @@
...
+
+  useEffect(() => {
+    if (userBounds !== "") {
+      getMarkers();
+    }
+  }, [userBounds]); // eslint-disable-line react-hooks/exhaustive-deps
```
Comment (christophersiem) [inline frontend/src/components/StreetMap.js:31]: Is the eslint-disable comment working here? Thought it has to be in line 31 ( before the dep. array of the hook)

File: frontend/src/components/StreetMap.js [hunk 7]
```diff
@@ -1,29 +1,84 @@
...
+      .then(setMarkers)
+      .catch((error) => console.log(error));
+  }
+
+
+  function PlayerController() {
     const map = useMapEvents({
       click: () => {
-        //Has to be changed to a timer
+        //ToDo: Has to be changed to a timer
```
Comment (christophersiem) [inline frontend/src/components/StreetMap.js]:
```suggestion

```

File: backend/src/main/java/de/cathesso/scratchamap/service/MapDataService.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+    public MapDataService(OverpassApiService apiService){this.apiService = apiService;}
+
+    public List<OverpassApiElement> getMapElements(String sWLat, String sWLon, String nELat, String nELon){
+        return apiService.getMapElements(sWLat, sWLon, nELat, nELon);
+    }
+
+    public List<Marker> getMarkers(String sWLat, String sWLon, String nELat, String nELon){
+        //Explanation: Nodes are all points from Overpass. Markers are all points that are to be shown (Overpass + future own generated points).
+        return apiService.getMapNodes(sWLat, sWLon, nELat, nELon);
+        //ToDo: Add own points for case: Way consists only of beginning and end --> Add point in middle.
```
Comment (christophersiem) [inline backend/src/main/java/de/cathesso/scratchamap/service/MapDataService.java]:
```suggestion

```

**Review by christophersiem — COMMENTED** [review]
Good work, but I'd remove all the system.out.println and comments before merging into main. Better keep track of your todo's withing your project planning :)

File: frontend/src/components/StreetMap.js (hunk 6 above)
Comment (Cathesso) [reply frontend/src/components/StreetMap.js:31]: Yup, the Github-Actions worked since I had this comment :)

**Pull request merged**